que.go

```diff
@@ -216,46 +216,56 @@ func (c *Client) LockJob(queue string) (*Job, error) {
 	}
 
 	j := Job{pool: c.pool, conn: conn}
-	err = conn.QueryRow("que_lock_job", queue).Scan(
-		&j.Queue,
-		&j.Priority,
-		&j.RunAt,
-		&j.ID,
-		&j.Type,
-		&j.Args,
-		&j.ErrorCount,
-	)
-	if err != nil {
-		c.pool.Release(conn)
-		if err == pgx.ErrNoRows {
-			return nil, nil
+
+	for {
```

Comment:
To eliminate the potential for an infinite loop, can you just change this into a `for i := 0; i < N; i++`, where `N` is some reasonable limit? I don't think it's reasonable for this to retry more than a small handful of times. If that's happening, something is probably wrong.

Then we can just return an exported error of some sort after the `for` loop to cover the repeat race condition edge case.

I understand that just picking a value like 2 or 3 is quite arbitrary, but this is supposed to be a rare edge case so the likelihood of it happening multiple times in a row is quite small. Still, I'd much rather return a rare error than potentially enter into an infinite loop :)